rotkehlchen/chain/zksync_lite/manager.py

```diff
@@ -801,18 +828,26 @@ def decode_transaction(
                 (1, transaction.tx_hash),
             )
 
-    def decode_undecoded_transactions(self) -> int:
-        """Decoded undecoded transactions.
+    def decode_undecoded_transactions(self, force_redecode: bool) -> int:
+        """Decodes undecoded zksync lite transactions. If force redecode is True
+        then all transactions are redecoded.
         Returns the number of decoded transactions (not events in transactions)
         """
-        transactions = self.get_db_transactions(
-            queryfilter=' WHERE is_decoded=?',
-            bindings=(0,),
-        )
+        queryfilter, bindings = '', ()
+        if not force_redecode:
+            queryfilter, bindings = ' WHERE is_decoded=?', (0,)  # type: ignore
+
+        transactions = self.get_db_transactions(queryfilter, bindings)
         with self.database.conn.read_ctx() as cursor:
             tracked_addresses = self.database.get_blockchain_accounts(cursor).zksync_lite
 
         for transaction in transactions:
+            with self.database.user_write() as write_cursor:  # delete old tx events
+                write_cursor.execute(
+                    'DELETE FROM history_events WHERE event_identifier=?',
+                    (f'zkl{transaction.tx_hash.hex()}',),
+                )
+
```

Comment:
here you have the cursor context being created and desttoyed on every loop iteration. Wouldn't it be better to do a executemany or a `delete WHERE event_identifier in [list comprehnesion to build f'zkl{transaction.tx_hash.hex()}' ]` and then do a loop only for self.decode_transaction(transaction, tracked_addresses). Yes you are looping twice but the user_write has a lot of logic querying the greenlet, checking savepoints, creating transactions...